pkg/querier/blocks_consistency_checker.go

```diff
@@ -113,6 +113,8 @@ func (c BlocksConsistencyTracker) Check(queriedBlocks []ulid.ULID) (missingBlock
 	return missingBlocks
 }
 
+// Complete should be called once the request tracked by this BlocksConsistencyTracker has been completed.
+// This function should NOT be called if the request is canceled or interrupted due to any error.
 func (c BlocksConsistencyTracker) Complete() {
```

Comment:
Note to reviewers: I changed the contract of this function. `Complete()` should be called only if the request was completed, and not cancelled or interrupted because an error occurred.

Reply:
I tried it at first, but the resulting design was pretty weird. `BlocksConsistencyTracker` has no knowledge of good/bad errors.